packages/react-storage/src/components/StorageBrowser/tasks/useProcessTasks.ts

```diff
@@ -36,19 +36,26 @@ export const useProcessTasks = <T, K>(
   React.useEffect(() => {
     if (!items?.length) return;
 
+    const createRemove = (targetId: string) =>
+      function remove() {
+        if (inflight.current.has(targetId)) return;
+
+        setTasks((prevTasks) => prevTasks.filter(({ id }) => id !== targetId));
+      };
+
     setTasks((prevTasks) => {
-      const nextTasks: Task<T>[] = items.reduce((tasks, item) => {
-        const remove = () => {
-          if (inflight.current.has(item.key)) return;
-
-          setTasks((prevTasks) =>
-            prevTasks.filter(({ key }) => key !== item.key)
-          );
-        };
-        return hasExistingTask(prevTasks, item)
-          ? tasks
-          : [...tasks, { ...item, ...QUEUED_TASK_BASE, remove }];
-      }, [] as Task<T>[]);
+      const nextTasks: Task<T>[] = items.reduce(
+        (tasks: Task<T>[], item) =>
+          prevTasks.some(({ id }) => id === item.id)
+            ? tasks
+            : [
+                ...tasks,
+                { ...item, ...QUEUED_TASK_BASE, remove: createRemove(item.id) },
+              ],
+        []
+      );
+
+      if (!nextTasks.length) return prevTasks;
```

Comment:
Should this be returning `[...prevTasks]`? If we don't need to return a copy, we may be able to optimize line 60 with `.concat`

Reply:
We want the reference here to remain stable since there is no change to `prevTasks` so no need to update state. Will make the `.concat` update , good catch